src/Tribe/Repositories/Event.php

```diff
@@ -1721,4 +1745,49 @@ public function filter_by_hidden_on_upcoming( $hidden ) {
 
 		$this->by( 'meta_not_exists', '_EventHideFromUpcoming' );
 	}
+
+	/**
+	 * Sets up the query filters to order events by the duration (`_EventDuration`) custom field.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $order      The order direction, either `ASC` or `DESC`; defaults to `null` to use the order
+	 *                           specified in the current query or default arguments.
+	 * @param bool   $after      Whether to append the duration ORDER BY clause to the existing clauses or not;
+	 *                           defaults to `false` to prepend the duration clause to the existing ORDER BY
+	 *                           clauses.
+	 * @param bool   $override   Whether to override existing ORDER BY clauses with this one or not; default to
+	 *                           `true` to override existing ORDER BY clauses.
+	 */
+	protected function order_by_duration( $order = null, $after = false, $override = true ) {
+		global $wpdb;
+
+		$meta_alias = 'event_duration';
+		$meta_key   = '_EventDuration';
+
+		$postmeta_table = "orderby_{$meta_alias}_meta";
+
+		$filter_id = 'order_by_duration';
+
+		$this->filter_query->join(
+			$wpdb->prepare(
+				"
+				LEFT JOIN {$wpdb->postmeta} AS {$postmeta_table}
+					ON (
+						{$postmeta_table}.post_id = {$wpdb->posts}.ID
+						AND {$postmeta_table}.meta_key = %s
+					)
+				",
+				$meta_key
+			),
+			$filter_id,
+			true
+		);
+
+		$order = $order === null
```

Comment:
:no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: d0f5d40b17a6e4dd730f324db4e3722a</sub>